packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -464,7 +464,8 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
     const minCapacity = props.minCapacity !== undefined ? props.minCapacity : 1;
     const maxCapacity = props.maxCapacity !== undefined ? props.maxCapacity : desiredCapacity;
 
-    if (desiredCapacity < minCapacity || desiredCapacity > maxCapacity) {
+    if ((!Token.isUnresolved(desiredCapacity) && !Token.isUnresolved(minCapacity) && !Token.isUnresolved(maxCapacity))
+    && (desiredCapacity < minCapacity || desiredCapacity > maxCapacity)) {
```

Comment:
You're losing a bit of coverage here. If `minCapacity` and `desiredCapacity` are both resolved, you can still check `desiredCapacity < minCapacity`.